tests/test1019.py

```diff
@@ -6,19 +6,50 @@
 ambrosia = sys.argv[1]
 font = fontforge.open(ambrosia)
 
-assert font.math.MathLeadingDeviceTable == ()
+assert font.math.MathLeadingDeviceTable == {}
 
+# Assign various adjustment sets
 for adjustments in [
-    ((9, -1), (10, -1), (12, -1)), # set non-empty
-    ((9, -1), (10, -1), (12, -1)), # set previous
-    ((9, -1), (10, -1)), # reduce
-    ((9, -1), (10, -1), (14, -1), (18, -1)), # increase
-    ((9, 1), (10, -2), (12, 0), (18, -3)), # change values
-    ((9, 1), (12, 0), (18, -3), (10, -2)), # change order
-    (), # set empty
-    () # set previous
+    {9: -1, 10: -1, 12: -1}, # set non-empty
+    {9: -1, 10: -1, 12: -1}, # set previous
+    {9: -1, 10: -1}, # reduce
+    {9: -1, 10: -1, 14: -1, 18: -1}, # increase
+    {9: 1, 10: -2, 12: 0, 18: -3}, # change values
+    {9: 1, 12: 0, 18: -3, 10: -2}, # change order
+    {}, # set empty
+    {} # set previous
 ]:
     font.math.MathLeadingDeviceTable = adjustments
-    adjustments = sorted(adjustments, key=lambda adj: adj[0])
-    adjustments = tuple((p, c) for p, c in adjustments if c != 0) # cor == 0 entries are not returned
-    assert font.math.MathLeadingDeviceTable == adjustments
+    # Filter out zero entries, they are not stored
+    non_zero_adjustments = {k:v for (k,v) in adjustments.items() if v != 0}
+    assert font.math.MathLeadingDeviceTable == non_zero_adjustments
+    assert len(font.math.MathLeadingDeviceTable) == len(non_zero_adjustments)
+
+# Set table to None
+font.math.AxisHeightDeviceTable = None
+assert font.math.AxisHeightDeviceTable == {}
```

Comment:
`font.math.AxisHeightDeviceTable` never contained values in this test, so if assigning `None` doesn't work (e.g. if it would have no effect), the test wouldn't catch it. Your implementation works fine, but to make the test check what it should check, you may want to assign a value before assigning `None`.
```suggestion
font.math.AxisHeightDeviceTable = {10: -1}
font.math.AxisHeightDeviceTable = None
assert font.math.AxisHeightDeviceTable == {}
```

Reply:
Right, done.